sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -145,3 +151,20 @@ def on_request(self, request):
             else:
                 url = url + "?" + signature
         request.http_request.url = url
+
+
+class AzureNamedKeyCredentialPolicy(SansIOHTTPPolicy):
+    """Adds a key header for the provided credential.
+
+    :param credential: The credential used to authenticate requests.
+    :type credential: ~azure.core.credentials.AzureNamedKeyCredential
+    :raises: ValueError or TypeError
+    """
+    def __init__(self, credential, **kwargs):  # pylint: disable=unused-argument
+        # type: (AzureNamedKeyCredential, **Any) -> None
+        super(AzureNamedKeyCredentialPolicy, self).__init__()
+        self._key = credential.key
```

Comment:
What happens when a client is created with a given `NamedKeyCredential` and later call `update` on the NamedKeyCredential? The whole reason for the update method is to be able to roll the key.